internals-js/src/specs/federationSpec.ts

```diff
@@ -181,6 +182,14 @@ export class FederationSpecDefinition extends FeatureDefinition {
     if (version.gte(new FeatureVersion(2, 9))) {
       this.registerSubFeature(COST_VERSIONS.find(new FeatureVersion(0, 1))!);
     }
+
+    if (version.gte(new FeatureVersion(2, 12))) {
+      this.registerDirective(createDirectiveSpecification({
+        name: FederationDirectiveName.CACHE_TAG,
+        locations: [DirectiveLocation.OBJECT, DirectiveLocation.INTERFACE, DirectiveLocation.FIELD_DEFINITION],
```

Comment:
Whats the use case for applying this on interface? We always have to return a concrete object types so whats the expected behavior here? If I apply it on interface should that be inherited by implementations?

```graphql
interface Product @cacheTag(format: "product-${id}") {
  id: ID!
  title: String!
}

type Book @key(fields: "id") {
  id: ID!
  # other fields
}

type Movie @key(fields: "id") {
  id: ID!
  # other fields
}
```

Reply:
Exactly, the semantics will be as if all implementors have the same cache tag definition.
The motivation to allow interface was consistency with `@key`. But, I realized that it should be only allowed if there IS a `@key` on interface.

Reply:
Added that limitation on the validation doc: https://docs.google.com/document/d/1oF28QJZGFbRRhGN4FBpD2Cbsrt2kecJkyoSrHzfjuUk/edit?usp=sharing

Reply:
Thanks. Yes, I added that to the limitation doc as a validation rule.